File: solutions/javascript/reverse-string/1/reverse-string.js
//
// Given a string, reverse it and return it
// For example: input: "cool" output: "looc"
//

export const reverseString = (string) => {
  // initialize result as empty string
  let reversed_string = '';
  // loop over length of string, back to front
  for (let i = string.length - 1; i >= 0; i--) {
    // append character to reversed string
    reversed_string += string[i];
  }
  // return reversed string to the caller
  return reversed_string;
};

File: solutions/cpp/reverse-string/1/reverse_string.cpp
#include "reverse_string.h"

namespace reverse_string {

// receiving a standard string and returning a standard string
std::string reverse_string(std::string str)
{
    // initialize reversed string
    std::string reversed_string = "";
    
    // loop over the length of the string, end to beginning, one charcter at a time
    for (int i = str.length() - 1; i >= 0; i--)
    {
        // append to reversed string
        reversed_string += str[i];
    }
    return reversed_string;
}

}  // namespace reverse_string
